backend/printlinear.ml

```diff
@@ -32,6 +32,36 @@ let instr' ?(print_reg = Printmach.reg) ppf i =
   let regsetaddr = Printmach.regsetaddr' ~print_reg in
   let test = Printmach.test' ~print_reg in
   let operation = Printmach.operation' ~print_reg in
+  if !Flambda_backend_flags.davail then begin
+    let module RAS = Reg_availability_set in
+    let ras_is_nonempty (set : RAS.t) =
+      match set with
+      | Ok set -> not (Reg_with_debug_info.Set.is_empty set)
+      | Unreachable -> true
+    in
```

Comment:
(Maybe `is_empty` could be defined in
`Reg_availability_set` and used here and
in `Printmach`.)

Reply:
Actually I'd prefer not to do that, because the semantics are a bit unclear.  Maybe when we've thought more about what these availability sets should really look like.